### PR TITLE
#543 Add frontend component for the page listing the submissions for a given assessment

### LINKED ISSUE #543: There needs to be a page that presents a list of submissions for an assessment
### Describe the Feature / Enhancement

The page will retrieve a list of all submissions from a particular assessment and display them to the user, giving them the ability to view a prior submission or create a new submission (if a participant).

The page will present this in a tabular form, similar to the layout of the assessments list page, but with details that are important to the context of viewing a submissions list instead.

This page would live at the path `/assessments/:assessmentId/submissions`.

### Additional Details and Resources

For program participants, it will retrieve all submissions from that particular assessment by that participant, and if the number of submissions by the participant is fewer than the maximum number of submissions for that assessment, and if the due date has not passed yet, it will give the program participant the ability to create a new submission.

For program facilitators, it will retrieve all program participants' submissions from that assessment and give the facilitator the option to view or grade that submission.

### Correct Project Selected

- [X] This issue is linked to the appropriate project

### Labels

- [X] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by kaiyolau**
## Proposed changes

This pull request resolves #543 by adding a page for the GET `/assessments/:assessmentId/submissions` route.

Mentee mode :
Screen Shot 2023-03-20 at 4 13 37 PM

Mentor mode :
Screen Shot 2023-03-20 at 4 13 47 PM

## Checklist

- [x] Related issue appears at beginning of pull request title with pound sign, and title describes the changes being proposed.
- [x] Related issue is linked in pull request description using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] The appropriate label has been chosen for this pull request.
- [x] The correct project has been selected for this pull request.
- [x] All commits in this branch, including merge commits, begin with the issue number and a pound sign.
- [x] Tests have been added, where appropriate; or, no tests are relevant for this pull request.
- [x] This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes.

### CODE REVIEW

**Comment by seidior** [comment]
Hi Kyle! Before we can review this, can you please pull the latest from GitHub (I updated your branch with the latest commits from the feature branch), run `yarn delint`, address the comments, and keep running `yarn delint` until you don't get any warnings or errors, then commit and push to GitHub? Thank you!

**Comment by seidior** [comment]
Hi @kaiyolau the force push I just did was to fix the ability to merge this with the current state of the feature branch; this means only your code should now be showing on the PR "Files Changed" page.

**Review by seidior — CHANGES_REQUESTED** [review]
Great work, Kyle! A couple of modifications I'd like to see, if you don't mind:

- In your PR description, you checked off "This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes." but your pull request clearly adds new UI to our webapp, so we'll need a screenshot.
- Can we move the "Mentor Mode" switch above the table (instead of beneath it) and align it to the right side of the container so that it appears above and to the top right of the table?
- In mentor mode, can we show the principal ID of the submissions?
- Finally, can we move the "Take a Quiz!" button to the right side of the container so it appears beneath and to the bottom right of the table? And can we change the text to "New Submission"?

**Comment by kaiyolau** [comment]
EOD : 
posting a screenshot for the mentor and mentee side to the pull request,
moved the mentor mode switch button to the right,
changed the name of the submission button to "New Submission" and moved it to the right.

**Comment by seidior** [comment]
Hi Kyle! Thanks for addressing my feedback. Rather than describing the final tweaks I'd like you to do, it may be more helpful to demonstrate via image:

submissions_page_facilitator

Do you mind making the appropriate changes to make this happen? The trickiest part will be the "facilitator mode" switch, which I solved when generating the image above with a nested `<Stack>` inside a `<Grid item>` (but I'm open to other solutions; though don't spend too much time on it, since it's only temporary for now anyway). You can keep the action item for participants as "Review"; only the action items for facilitators should change to the above.

**Comment by kaiyolau** [comment]
EOD : 

today I have done adding more conditions for the button within the "Action" column, 
and apply "Chip" styling for the "State" column,
also moved the "student ID" column to the first column.

**Review by seidior — APPROVED** [review]
Hi @kaiyolau! Thank you for incorporating my feedback from my previous review. Everything looks great here! The page renders beautifully, and the outcome here was even better than I'd hoped.

(I did have to temporarily change the due date for the example assessment on my system so that I could see the "New Submission" button, but that's not your fault!)

I had some extra time this morning so I brought your branch up to date with the latest changes from the feature branch so that you're not roadblocked from merger based on that. Once you have two approving reviews, you're welcome to merge!

File: webapp/src/components/AssessmentSubmissionsListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    fontSize: 14,
+    textAlign: 'center',
+  },
+}));
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  '&:nth-of-type(odd)': {
+    backgroundColor: theme.palette.action.hover,
+  },
+  // hide last border
```
Comment (mays4) [inline webapp/src/components/AssessmentSubmissionsListPage.tsx]: Remove comment

File: webapp/src/components/AssessmentSubmissionsListPage.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+  '&:last-child td, &:last-child th': {
+    border: 0,
+  },
+}));
+
+const submissionsExample = [
```
Comment (mays4) [inline webapp/src/components/AssessmentSubmissionsListPage.tsx]: Why you don't move this submissionsExample to data file ?

**Review by mays4 — APPROVED** [review]
Great work , I like the new features added to render based in the role

**Pull request merged**